### ISSUE TITLE
StrawberryShake.Blazor causes thousands of build errors

### ISSUE THREAD

**Issue by holotrek** [ChilliCream/graphql-platform#6211]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Product

Strawberry Shake

### Describe the bug

Just creating a simple Blazor WASM project in .NET 7 and adding StrawberryShake.Blazor causes a bunch of errors when I try to build it. The first thing that happens is that I get a warning, and if I ctrl+c the console, I'm able to copy that warning. It seems to indicate that the build is enumerating my ENTIRE DRIVE, which is crazy:

```
MSBUILD : warning MSB5029: The value "\**\*.cs" of the "Include" attribute in element <ItemGroup> in file "C:\Users\evshe\.nuget\packages\strawberryshake.blazor\13.1.0\build\Straw
berryShake.Blazor.targets (23,16)" is a wildcard that results in enumerating all files on the drive, which was likely not intended. Check that referenced properties are always def 
ined. [C:\Users\evshe\source\repos\TestBlazorGraphQL\TestBlazorGraphQL.csproj]
```

Also, when I added it originally in VS2022, it started downloading tons of files from OneDrive into a Unity folder which was in a completely separate place than this project. It froze up VS and I had to completely redo a lot of things. My guess is that this drive enumeration is causing it to try to download ALL my OneDrive files, which is not acceptable. The only way I was able to safely reproduce the issue was in VS Code with CLI commands.

### Steps to reproduce

1. Create a new Blazor WASM project and add StrawberryShake.Blazor, then build it:
```
dotnet new blazorwasm -o TestBlazorGraphQL
cd TestBlazorGraphQL
dotnet add package StrawberryShake.Blazor
dotnet build
```

### Relevant log output

```shell
Warning (if Ctrl+c):

MSBUILD : warning MSB5029: The value "\**\*.cs" of the "Include" attribute in element <ItemGroup> in file "C:\Users\evshe\.nuget\packages\strawberryshake.blazor\13.1.0\build\StrawberryShake.Blazor.targets (23,16)" is a wildcard that results in enumerating all files on the drive, which was likely not intended. Check that referenced properties are always defined. [C:\Users\evshe\source\repos\TestBlazorGraphQL\TestBlazorGraphQL.csproj]

End of error output if I let it go:

C:\Program Files\Microsoft Visual Studio\2022\Community\Common7\IDE\ItemTemplates\Web\CSharp\1033\DynamicDataWebSiteField.cs\DynamicData\FieldTemplates\CodeBeside_Edit.cs(37,1): error CS1022: Type or namespace definition, or end-of-file expected [C:\Users\evshe\source\repos\TestBlazorGraphQL\TestBlazorGraphQL.csproj]
    5 Warning(s)
    15077 Error(s)
```

### Version

13.1.0

**Comment by holotrek** [comment]
I decided to take a look at that StrawberryShake.Blazor.targets file and noticed a couple lines that were very weird and suspicious:
```
<Compile Include="$(GraphQLCodeGenerationRoot)\**\*.cs" />
...
<GraphQLCodeGenerationRoot>$([System.IO.Path]::GetFullPath('$(IntermediateOutputPath)berry'))</GraphQLCodeGenerationRoot>
```

I feel like `$(IntermediateOutputPath)berry` is not a real path that would cause the generation root to be empty, thus resulting in the full drive enumeration. I tried clearing my NuGet cache, which ended up deleting that file, but the rebuild created it again with the same unusual lines.

It also happens with all the following versions:
- 13.0.0
- 13.0.4
- 13.0.5
- 13.1.0
- 14.0.0-preview.11.3

**Comment by orlandommb** [comment]
im having this issue as well !

**Comment by holotrek** [comment]
I figured out the issue. You actually have to generate the graphql files following [this tutorial](https://chillicream.com/docs/strawberryshake/v13/get-started) BEFORE building. I feel like this needs a safeguard.

Also, that ended up being a pain, because I'm using a ASP.NET hosted version of my client, from the MudBlazor `Wasm-PWA-Hosted` template. This adds the Client project as a reference in the Server, so I had to temporarily remove this reference in order to start my server so that `dotnet graphql init` would work.

**Comment by holotrek** [comment]
OK, apparently having to generate the graphql files is a red herring. It seems that the `_GenerateGraphQLCode` target only runs in the following situations:
- There's something named `.graphql` in the project
- Something has changed with one of the `.graphql` files OR a change to `obj` was detected

Therefore, the build seems to work partway, but then fails literally by running `dotnet build` again a 2nd time. This is because it skips the  `_GenerateGraphQLCode` target.

This was reproduced by doing the following:
1. Create a new Blazor WASM project:
```
dotnet new blazorwasm -o TestBlazorGraphQL
cd TestBlazorGraphQL
```
2. Add `any.graphql` to the project (you don't even have to put anything in it)
3. Add StrawberryShake.Blazor (DO NOT build yet):
```
dotnet add package StrawberryShake.Blazor
```
4. DELETE both the /bin and /obj directories
5. Build the first time (succeeds and output shows "Generate C# Clients" target running)
```
dotnet build
```
6. Build the 2nd time (fails with an error about the .NET version and the warning about the Include attribute enumerating all files on the drive)
```
dotnet build
```

I discovered this by running the build with diagnostics and viewing the resulting file:
```
dotnet build /v:diag | Out-File -FilePath ./build-output.txt
```

Note that the output contains the following when no changes have been detected:
```
Skipping target "_GenerateGraphQLCode" because all output files are up-to-date with respect to the input files.
                   Input files: 
                       any.graphql
                       C:\Users\evshe\source\repos\TestBlazorGraphQL\TestBlazorGraphQL.csproj
                       C:\Users\evshe\.nuget\packages\strawberryshake.blazor\13.1.0\build\StrawberryShake.Blazor.targets
```

Skipping this target is bad because then it doesn't create the `$(GraphQLCodeGenerationRoot)` variable (does not even come up in a search in diagnostics build output). When I delete the /bin and /obj and run it with diagnostics again, I can see the value of that variable:
```
Set Property: GraphQLCodeGenerationRoot=C:\Users\evshe\source\repos\TestBlazorGraphQL\obj\Debug\net7.0\berry
```

**Comment by orlandommb** [comment]
i've removed everything graphql related from my client side, and followed the tutorial again, also we gotta update the Strawberry.Shake Tools, im still debuggin my project, i'll get back to report any findings !

**Comment by holotrek** [comment]
Just to rule out any possible environmental issues with my machine, I used docker and got the same result (I also added the tool manifest and the Strawberry Shake tools, but it really doesn't make any difference). Run from the app directory:
```
docker run --rm -v ${PWD}:/app mcr.microsoft.com/dotnet/sdk:7.0 dotnet build /app
```

Same issue, although the errors were a lot smaller, since it is a container:
```
user@machine TestBlazorGraphQL % docker run --rm -v ${PWD}:/app mcr.microsoft.com/dotnet/sdk:7.0 dotnet build /app   
MSBuild version 17.4.0+18d5aef85 for .NET
  Determining projects to restore...
  Restored /app/TestBlazorGraphQL.csproj (in 5.09 sec).
  Generate C# Clients started.
  Generate C# Clients completed in 57 ms
  TestBlazorGraphQL -> /app/bin/Debug/net7.0/TestBlazorGraphQL.dll
  TestBlazorGraphQL (Blazor output) -> /app/bin/Debug/net7.0/wwwroot

Build succeeded.
    0 Warning(s)
    0 Error(s)

Time Elapsed 00:00:15.49
user@machine TestBlazorGraphQL % docker run --rm -v ${PWD}:/app mcr.microsoft.com/dotnet/sdk:7.0 dotnet build /app
MSBuild version 17.4.0+18d5aef85 for .NET
  Determining projects to restore...
  Restored /app/TestBlazorGraphQL.csproj (in 4.64 sec).
/root/.nuget/packages/strawberryshake.blazor/13.1.0/build/StrawberryShake.Blazor.targets(38,20): error MSB4184: The expression "[MSBuild]::VersionGreaterThanOrEquals('', 6)" cannot be evaluated. Version string was not in a correct format. [/app/TestBlazorGraphQL.csproj]
MSBUILD : warning MSB5029: The value "\**\*.cs" of the "Include" attribute in element <ItemGroup> in file "/root/.nuget/packages/strawberryshake.blazor/13.1.0/build/StrawberryShake.Blazor.targets (23,16)" is a wildcard that results in enumerating all files on the drive, which was likely not intended. Check that referenced properties are always defined. [/app/TestBlazorGraphQL.csproj]
CSC : error CS2021: File name '/**/*.cs' is empty, contains invalid characters, has a drive specification without an absolute path, or is too long [/app/TestBlazorGraphQL.csproj]
CSC : error CS2001: Source file '/**/*.cs' could not be found. [/app/TestBlazorGraphQL.csproj]

Build FAILED.

MSBUILD : warning MSB5029: The value "\**\*.cs" of the "Include" attribute in element <ItemGroup> in file "/root/.nuget/packages/strawberryshake.blazor/13.1.0/build/StrawberryShake.Blazor.targets (23,16)" is a wildcard that results in enumerating all files on the drive, which was likely not intended. Check that referenced properties are always defined. [/app/TestBlazorGraphQL.csproj]
/root/.nuget/packages/strawberryshake.blazor/13.1.0/build/StrawberryShake.Blazor.targets(38,20): error MSB4184: The expression "[MSBuild]::VersionGreaterThanOrEquals('', 6)" cannot be evaluated. Version string was not in a correct format. [/app/TestBlazorGraphQL.csproj]
CSC : error CS2021: File name '/**/*.cs' is empty, contains invalid characters, has a drive specification without an absolute path, or is too long [/app/TestBlazorGraphQL.csproj]
CSC : error CS2001: Source file '/**/*.cs' could not be found. [/app/TestBlazorGraphQL.csproj]
    1 Warning(s)
    3 Error(s)

Time Elapsed 00:00:07.23
```

**Comment by holotrek** [comment]
For those who are interested, I found a workaround. After adding the StrawberryShake.Blazor package, find the targets file for that version at: `%USERPROFILE%/.nuget/packages/strawberryshake.blazor/{VERSION}/build/StrawberryShake.Blazor.targets`

Find the target below and remove the "Inputs" attribute:
```
  <Target
    Name="_GenerateGraphQLCode"
    Inputs="@(GenerateGraphQLCodeItems)"
    Outputs="$(MSBuildProjectDirectory)\$(IntermediateOutputPath)berry\.build.info"
    DependsOnTargets="_ParseVersion"
    Condition="'@(GraphQL)' != ''">
```

This will force it to regenerate the client every time and thus properly create the `$(GraphQLCodeGenerationRoot)` variable.

**Comment by michaelstaib** [comment]
@holotrek thanks for the analysis, this saved me a lot of time in creating a fix which should be available with the next 13.2 preview.
